File: 프로그래머스/1/42840. 모의고사/모의고사.py
def solution(answers):
    people_1 = [1, 2, 3, 4, 5]
    people_2 = [2, 1, 2, 3, 2, 4, 2, 5]
    people_3 = [3, 3, 1, 1, 2, 2, 4, 4, 5, 5]

    temp = [0, 0, 0]
    for i in range(len(answers)):

        if people_1[i % len(people_1)] == answers[i]:
            temp[0] += 1
        if people_2[i % len(people_2)] == answers[i]:
            temp[1] += 1
        if people_3[i % len(people_3)] == answers[i]:
            temp[2] += 1

    target = max(temp)
    answer = []
    for i in range(len(temp)):
        if temp[i] == target:
            answer.append(i+1)

    return answer

File: 프로그래머스/1/42840. 모의고사/모의고사.java
import java.util.*;

class Solution {
    static int[] A = new int[]{1, 2, 3, 4, 5}; // 5
    static int[] B = new int[]{2, 1, 2, 3, 2, 4, 2, 5}; // 8
    static int[] C = new int[]{3, 3, 1, 1, 2, 2, 4, 4, 5, 5}; // 10
    public int[] solution(int[] answers) {
        int[] res = new int[3];
        for (int i = 0; i < answers.length; i++){
            if(answers[i] == A[i%5]){
                res[0]++;
            }
            if(answers[i] == B[i%8]){
                res[1]++;
            }
            if(answers[i] == C[i%10]){
                res[2]++;
            }
        }
        int top = 0;
        for(int i : res){
            top = Math.max(top, i);
        }
        List<Integer> arr = new ArrayList<>();
        for(int i = 0; i < res.length; i++){
            if(res[i] == top){
                arr.add(i+1);
            }
        }
        arr.sort((a,b)->a-b);
        return arr.stream().mapToInt(i->i).toArray();
    }
}